### PR TITLE
date filter added

### PR DESCRIPTION
**Pull request by BINAH25**
date filter added to the tables
image
image

### CODE REVIEW

File: backend/dashboard/migrations/0105_alter_facility_options_and_more.py
Comment (dodziraynard) [inline backend/dashboard/migrations/0105_alter_facility_options_and_more.py:1]: Why this file?

File: backend/rest_api/views/web/dashboard_charts.py [hunk 1]
```diff
@@ -178,7 +178,12 @@ class BasicDemographics(generics.GenericAPIView):
 
     @method_decorator(cache_page(60 * 2))
     def get(self, request, format=None):
-        response_data = get_demographic_data(adolescent_type="basic")
+        start_date = _get_date(request.GET.get("start_date")) or make_aware(
+            datetime(2023, 1, 1))
+        end_date = _get_date(request.GET.get("end_date")) or timezone.now()
+        response_data = get_demographic_data(adolescent_type="basic",start_date=start_date,end_date=end_date)
+        print(response_data)
```
Comment (dodziraynard) [inline backend/rest_api/views/web/dashboard_charts.py:185]:
```suggestion
```

File: backend/rest_api/views/web/dashboard_charts.py [hunk 2]
```diff
@@ -196,7 +204,11 @@ class CommunityDemographics(generics.GenericAPIView):
 
     @method_decorator(cache_page(60 * 2))
     def get(self, request, format=None):
-        response_data = get_demographic_data(adolescent_type="community")
+        start_date = _get_date(request.GET.get("start_date")) or make_aware(
+            datetime(2023, 1, 1))
+        end_date = _get_date(request.GET.get("end_date")) or timezone.now()
+        response_data = get_demographic_data(adolescent_type="community",start_date=start_date,end_date=end_date)
+        print(response_data)
```
Comment (dodziraynard) [inline backend/rest_api/views/web/dashboard_charts.py:211]:
```suggestion
```

Comment (BINAH25) [reply backend/rest_api/views/web/dashboard_charts.py]: removed

File: backend/rest_api/views/web/dashboard_charts.py (hunk 1 above)
Comment (BINAH25) [reply backend/rest_api/views/web/dashboard_charts.py]: removed

**Review by dodziraynard — APPROVED** [review]

**Pull request merged**